src/lib.rs

```diff
@@ -66,6 +74,12 @@ impl<T: FileLoader> VfsInternal<T> {
         }
     }
 
+    fn load_file(&self, path: &Path) {
+        let file = T::read(path).unwrap();
+        let mut files = self.files.lock().unwrap();
+        files.insert(path.to_path_buf(), file);
```

Comment:
we should assert that the file has not yet been loaded (otherwise this could overwrite changes on an unsaved file)